### PR TITLE
MPP: fix 2-phase agg chose wrong partition column during planning

### PR DESCRIPTION
**Pull request by fzhedu**
### What problem does this PR solve?

Issue Number: close https://github.com/pingcap/tics/issues/1648 
introduced by plan/core: support mpp group by expressions. (#23133)

Problem Summary:
the parent's partition columns properity in exhaust_plan is removed during `attach2TaskForMpp`

### What is changed and how it works?

Proposal: [xxx](url) 

What's Changed: recover the partition column properity from parent

How it Works:

### Related changes

- Need to cherry-pick to the release branch

### Check List 

Tests 

- Unit test
- Integration test
- Manual test (add detailed scripts or steps below)

### Release note 

- No release note

### CODE REVIEW

File: planner/core/task.go [hunk 1]
```diff
@@ -1597,16 +1597,18 @@ func (p *PhysicalHashAgg) attach2TaskForMpp(tasks ...task) task {
 			return invalidTask
 		}
 		attachPlan2Task(partialAgg, mpp)
-		items := finalAgg.(*PhysicalHashAgg).GroupByItems
-		partitionCols := make([]*expression.Column, 0, len(items))
-		for _, expr := range items {
-			col, ok := expr.(*expression.Column)
-			if !ok {
-				return invalidTask
+		if len(p.MppPartitionCols) == 0 {
```
Comment (hanfei1991) [inline planner/core/task.go:1600]: "attach2TaskForMpp" will be called mutiple times. Everytime the group-by columns will have a unique ID. So please construct it everytime.

File: planner/core/testdata/integration_serial_suite_out.json [hunk 2]
```diff
@@ -1776,6 +1776,43 @@
...
+      },
+      {
+        "SQL": "desc format = 'brief' select /*+hash_agg()*/ sum(id) from (select value, id from t where id > value group by id, value)A group by value /*the exchange should have only one partition column: test.t.value*/",
+        "Plan": [
+          "TableReader 6400.00 root  data:ExchangeSender",
+          "└─ExchangeSender 6400.00 batchCop[tiflash]  ExchangeType: PassThrough",
+          "  └─Projection 6400.00 batchCop[tiflash]  Column#4",
+          "    └─HashAgg 6400.00 batchCop[tiflash]  group by:test.t.value, funcs:sum(test.t.id)->Column#4",
+          "      └─Projection 6400.00 batchCop[tiflash]  test.t.id, test.t.value",
+          "        └─HashAgg 6400.00 batchCop[tiflash]  group by:test.t.id, test.t.value, funcs:firstrow(test.t.id)->test.t.id, funcs:firstrow(test.t.value)->test.t.value",
```
Comment (hanfei1991) [inline planner/core/testdata/integration_serial_suite_out.json:1788]: The lower Agg doesn't choose a two phase agg, so this doesn't really cover your case.

Comment (hanfei1991) [reply planner/core/testdata/integration_serial_suite_out.json:1788]: But the next test covers it, then that's ok.

File: planner/core/task.go (hunk 1 above)
Comment (winoros) [reply planner/core/task.go:1600]: Why the uniqueID will change every time?

**Review by qw4990 — APPROVED** [review]

**Review by hanfei1991 — APPROVED** [review]

**Pull request merged**